astropy/io/fits/_tiled_compression/utils.py

```diff
@@ -31,3 +31,62 @@ def _iter_array_tiles(data_shape, tile_shape):
 
         if istart[0] >= data_shape[0]:
             break
+
+
+def _tile_index_to_row_index(tile_index, n_tiles):
+    row_index = tile_index[0]
+    for dim in range(1, len(tile_index)):
+        row_index = tile_index[dim] + row_index * n_tiles[dim]
+    return row_index
+
+
+def _tile_index_to_tile_slices(tile_index, tile_shape):
+    return tuple(
+        [
+            slice(
+                tile_index[idx] * tile_shape[idx],
+                (tile_index[idx] + 1) * tile_shape[idx],
+            )
+            for idx in range(len(tile_index))
+        ]
+    )
+
+
+def _n_tiles(data_shape, tile_shape):
+    return tuple(int(np.ceil(d / t)) for d, t in zip(data_shape, tile_shape))
+
+
+def _iter_array_tiles_subset(data_shape, tile_shape, first_tile_index, last_tile_index):
```

Comment:
If we make first_tile_index and last_tile_index optional we might be able to use only this and remove ``_iter_array_tiles``

Reply:
Done